sql/core/src/test/resources/sql-tests/results/datetime.sql.out

```diff
@@ -1013,3 +1013,110 @@ select date_format('2020-01-01', 'YYYY-MM-dd uuuu')
+ (timestamp '2018-11-17 13:33:33.123Z'),
+ (timestamp '2020-01-01 01:33:33.123Asia/Shanghai'),
+ (timestamp '2100-01-01 01:33:33.123America/Los_Angeles') tt(t)
+-- !query schema
+struct<>
+-- !query output
+
+
+
+-- !query
+select date_format(t, 'Y YY YYYY YYYYY YYYYY y yy yyy yyyy yyyyy') from ttt
+-- !query schema
+struct<date_format(t, Y YY YYYY YYYYY YYYYY y yy yyy yyyy yyyyy):string>
+-- !query output
+1582 82 1582 01582 01582 1582 82 1582 1582 01582
+1970 70 1970 01970 01970 1969 69 1969 1969 01969
+1970 70 1970 01970 01970 1970 70 1970 1970 01970
+1996 96 1996 01996 01996 1996 96 1996 1996 01996
+2018 18 2018 02018 02018 2018 18 2018 2018 02018
+2020 20 2020 02020 02020 2019 19 2019 2019 02019
+2099 99 2099 02099 02099 2100 00 2100 2100 02100
+
+
+-- !query
+select date_format(t, 'q qq Q QQ QQQ QQQQ') from ttt
+-- !query schema
+struct<date_format(t, q qq Q QQ QQQ QQQQ):string>
+-- !query output
+1 01 1 01 Q1 1st quarter
+1 01 1 01 Q1 1st quarter
+2 02 2 02 Q2 2nd quarter
+4 04 4 04 Q4 4th quarter
+4 04 4 04 Q4 4th quarter
+4 04 4 04 Q4 4th quarter
+4 04 4 04 Q4 4th quarter
+
+
+-- !query
+select date_format(t, 'M MM MMM MMMM L LL') from ttt
+-- !query schema
+struct<date_format(t, M MM MMM MMMM L LL):string>
+-- !query output
+1 01 Jan January 1 01
+11 11 Nov November 11 11
+12 12 Dec December 12 12
+12 12 Dec December 12 12
+12 12 Dec December 12 12
+3 03 Mar March 3 03
+5 05 May May 5 05
+
+
+-- !query
+select date_format(t, 'W ww d dd DDD u uu uuu uuuu F E EE EEE EEEE a') from ttt
+-- !query schema
+struct<date_format(t, W ww d dd DDD u uu uuu uuuu F E EE EEE EEEE a):string>
+-- !query output
+0 53 1 01 001 5 05 Fri Friday 1 Fri Fri Fri Friday AM
+3 46 17 17 321 6 06 Sat Saturday 3 Sat Sat Sat Saturday AM
+4 13 31 31 091 7 07 Sun Sunday 3 Sun Sun Sun Sunday AM
+5 01 31 31 365 2 02 Tue Tuesday 3 Tue Tue Tue Tuesday AM
+5 01 31 31 365 3 03 Wed Wednesday 3 Wed Wed Wed Wednesday PM
+5 22 31 31 151 1 01 Mon Monday 3 Mon Mon Mon Monday PM
+5 53 31 31 365 4 04 Thu Thursday 3 Thu Thu Thu Thursday AM
+
+
+-- !query
+select date_format(t, 'h hh H HH k kk K KK m mm s ss SSS') from ttt
+-- !query schema
+struct<date_format(t, h hh H HH k kk K KK m mm s ss SSS):string>
+-- !query output
+1 01 1 01 1 01 1 01 33 33 33 33 123
+3 03 15 15 15 15 3 03 0 00 0 00 000
+4 04 4 04 4 04 4 04 59 59 59 59 999
+5 05 5 05 5 05 5 05 33 33 33 33 123
+7 07 19 19 19 19 7 07 40 40 35 35 123
+7 07 7 07 7 07 7 07 3 03 33 33 123
+9 09 9 09 9 09 9 09 33 33 33 33 123
+
+
+-- !query
+select date_format(t, 'VV z zz zzz zzzz O OOOO X XX XXX XXXX XXXXX x xx xxx xxxx xxxx xxxxx Z ZZ ZZZ ZZZZ ZZZZZ') from ttt
+-- !query schema
+struct<date_format(t, VV z zz zzz zzzz O OOOO X XX XXX XXXX XXXXX x xx xxx xxxx xxxx xxxxx Z ZZ ZZZ ZZZZ ZZZZZ):string>
+-- !query output
+America/Los_Angeles PST PST PST Pacific Standard Time GMT-7:52:58 GMT-07:52:58 -0752 -0752 -07:52 -075258 -07:52:58 -0752 -0752 -07:52 -075258 -075258 -07:52:58 -0752 -0752 -0752 GMT-07:52:58 -07:52:58
```

Comment:
Do you know what causes this inaccuracy for dates before 1582-10-15？@cloud-fan

Reply:
I'm not sure, but it seems like we should ban all the timezone letters during formatting. Spark's timestamp value doesn't have an associated timezone, it makes no sense to extract timezone from it.

Reply:
cc @MaxGekk

Reply:
It seems to only display the value of `spark.sql.session.timeZone`

Reply:
actually, this is expected. The Spark test uses timezone `America/Los_Angeles`, which is indeed GMT-7:52:58 long time before, see https://www.politico.com/story/2017/11/18/railroads-initiative-synchronizes-us-clocks-nov-18-1883-244935

Spark's timestamp value doesn't have an associated timezone, so extracting timezone means showing the session local timezone.